scripts/run-clang-tidy-on-compile-commands.py

```diff
@@ -207,16 +207,26 @@ def Failure(self, path: str):
 
 
 def find_darwin_gcc_sysroot():
-    for line in subprocess.check_output(
-        "xcodebuild -sdk -version".split(), text=True
-    ).splitlines():
-        if not line.startswith("Path: "):
-            continue
-        path = line[line.find(": ") + 2:]
-        if "/MacOSX.platform/" not in path:
-            continue
-        logging.info("Found %s" % path)
-        return path
+    try:
+        for line in subprocess.check_output(
+            "xcodebuild -sdk -version".split(), text=True
+        ).splitlines():
+            if not line.startswith("Path: "):
+                continue
+            path = line[line.find(": ") + 2:]
+            if "/MacOSX.platform/" not in path:
+                continue
+            logging.info("Found %s" % path)
+            return path
+    except Exception:
+        # lets try with xcrun
+        try:
+            path = subprocess.check_output(
+                "xcrun --sdk macosx --show-sdk-path".split(), text=True
```

Comment:
Should we just switch to this always? Or are we concerned not to regress anything?

Reply:
We could... `xcrun` will pop up a UI dialog prompting the user to install the Xcode developer tools if there is no active Xcode or command line tools install. So it may be better to stick with trying xcodebuild first.

Note the xcodebuild logic can be simplified by running `xcodebuild -sdk macosx -version` instead of searching through all the platforms.

Reply:
I have changed that logic to `xcodebuild -sdk macosx -version` and removed an if check for `"/MacOSX.platform/"`. Probably will be tested in CI.